src/material-table.js

```diff
@@ -503,8 +503,23 @@ export default class MaterialTable extends React.Component {
     });
   }
 
+  onExpandedHook = (rowData) => {
+    const isHookDefined = !!this.props.onDetailPanelExpanded;
+    if (isHookDefined) {
```

Comment:
can't you just do 

```
if(this.props.onDetailPanelExpanded)
```

Reply:
I could. Thought this is more readable, but happy to change it if need be.